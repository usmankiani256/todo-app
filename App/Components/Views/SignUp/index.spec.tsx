import * as React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import SignUpScreen from './index'

describe('SignUpScreen Component', () => {
  it('renders correctly', () => {
    const props: any = {}

    const rendered: any = render(<SignUpScreen {...props} />).toJSON()

    expect(rendered).toMatchSnapshot()
  })

  it('navigates to Sign In screen', async () => {
    const props: any = {
      navigation: {
        navigate: jest.fn(),
      },
    }

    const { getByTestId } = render(<SignUpScreen {...props} />)

    fireEvent.changeText(getByTestId('sign-up-input-name'), 'test user')
    fireEvent.changeText(getByTestId('sign-up-input-email'), 'test@user.com')
    fireEvent.changeText(getByTestId('sign-up-input-password'), 'test')

    fireEvent.press(getByTestId('sign-up-button-create'))

    await new Promise(r => setTimeout(r, 2000))

    expect(props.navigation.navigate).toBeCalledWith('SignIn', {
      email: 'test@user.com',
    })
  })
})
