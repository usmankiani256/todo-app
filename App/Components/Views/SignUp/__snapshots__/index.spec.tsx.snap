// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpScreen Component renders correctly 1`] = `
<RCTScrollView
  contentInsetAdjustmentBehavior="automatic"
  keyboardShouldPersistTaps="handled"
  scrollEnabled={false}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexGrow": 1,
    }
  }
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "height": 1334,
              "width": 750,
            }
          }
        >
          <View
            style={
              Object {
                "left": 10,
                "position": "absolute",
                "top": 10,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18.75,
                      "height": 37.5,
                      "width": 37.5,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#3f3f3f",
                        "fontSize": 25,
                      },
                      Array [
                        Object {
                          "lineHeight": 25,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  Û∞ÖÅ
                </Text>
              </View>
            </View>
          </View>
          <Image
            resizeMethod="resize"
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../App/Assets/Images/logo.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "marginTop": 133.4,
                "width": 250,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#31394F",
                "fontFamily": "SofiaPro-Medium",
                "fontSize": 25,
                "lineHeight": 25,
                "paddingHorizontal": 18,
                "textAlign": "center",
              }
            }
          >
            Sign Up
          </Text>
          <Text
            style={
              Object {
                "color": "#777b",
                "fontFamily": "SofiaPro-Light",
                "fontSize": 17.5,
                "lineHeight": 17.5,
                "marginTop": 10,
                "paddingHorizontal": 18,
                "textAlign": "center",
              }
            }
          >
            Create an account and start using the application
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginTop": 20,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgb(231, 231, 231)",
                    "borderTopLeftRadius": 4,
                    "borderTopRightRadius": 4,
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "elevation": 1,
                    "marginTop": 15,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.18,
                    "shadowRadius": 1,
                    "width": 714,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "bottom": 0,
                    "height": 2,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "transform": Array [
                      Object {
                        "scaleY": 0.5,
                      },
                    ],
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "minHeight": 52,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  nativeID="animatedComponent"
                  pointerEvents="none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <Text
                    collapsable={false}
                    nativeID="animatedComponent"
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      Object {
                        "color": "#6200ee",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 28,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Full Name
                  </Text>
                  <Text
                    collapsable={false}
                    nativeID="animatedComponent"
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 28,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Full Name
                  </Text>
                </View>
                <TextInput
                  editable={true}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  outlineColor="#FFFFFF"
                  placeholder=""
                  placeholderTextColor="rgba(0, 0, 0, 0.54)"
                  selectionColor="#6200ee"
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "margin": 0,
                        "zIndex": 1,
                      },
                      Object {
                        "paddingLeft": 12,
                        "paddingRight": 12,
                      },
                      Object {
                        "height": 52,
                      },
                      Object {
                        "paddingBottom": 2,
                        "paddingTop": 22,
                      },
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      Array [
                        Object {},
                      ],
                    ]
                  }
                  testID="sign-up-input-name"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgb(231, 231, 231)",
                    "borderTopLeftRadius": 4,
                    "borderTopRightRadius": 4,
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "elevation": 1,
                    "marginTop": 15,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.18,
                    "shadowRadius": 1,
                    "width": 714,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "bottom": 0,
                    "height": 2,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "transform": Array [
                      Object {
                        "scaleY": 0.5,
                      },
                    ],
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "minHeight": 52,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  nativeID="animatedComponent"
                  pointerEvents="none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <Text
                    collapsable={false}
                    nativeID="animatedComponent"
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      Object {
                        "color": "#6200ee",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 28,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Email Address
                  </Text>
                  <Text
                    collapsable={false}
                    nativeID="animatedComponent"
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 28,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Email Address
                  </Text>
                </View>
                <TextInput
                  editable={true}
                  keyboardType="email-address"
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  outlineColor="#FFFFFF"
                  placeholder=""
                  placeholderTextColor="rgba(0, 0, 0, 0.54)"
                  selectionColor="#6200ee"
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "margin": 0,
                        "zIndex": 1,
                      },
                      Object {
                        "paddingLeft": 12,
                        "paddingRight": 12,
                      },
                      Object {
                        "height": 52,
                      },
                      Object {
                        "paddingBottom": 2,
                        "paddingTop": 22,
                      },
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      Array [
                        Object {},
                      ],
                    ]
                  }
                  testID="sign-up-input-email"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgb(231, 231, 231)",
                    "borderTopLeftRadius": 4,
                    "borderTopRightRadius": 4,
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "elevation": 1,
                    "marginTop": 15,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.18,
                    "shadowRadius": 1,
                    "width": 714,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "bottom": 0,
                    "height": 2,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "transform": Array [
                      Object {
                        "scaleY": 0.5,
                      },
                    ],
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "minHeight": 52,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  nativeID="animatedComponent"
                  pointerEvents="none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <Text
                    collapsable={false}
                    nativeID="animatedComponent"
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      Object {
                        "color": "#6200ee",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingLeft": 12,
                        "paddingRight": 44,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 28,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Password
                  </Text>
                  <Text
                    collapsable={false}
                    nativeID="animatedComponent"
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingLeft": 12,
                        "paddingRight": 44,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 28,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Password
                  </Text>
                </View>
                <TextInput
                  editable={true}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  outlineColor="#FFFFFF"
                  placeholder=""
                  placeholderTextColor="rgba(0, 0, 0, 0.54)"
                  secureTextEntry={true}
                  selectionColor="#6200ee"
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "margin": 0,
                        "zIndex": 1,
                      },
                      Object {
                        "paddingLeft": 12,
                        "paddingRight": 44,
                      },
                      Object {
                        "height": 52,
                      },
                      Object {
                        "paddingBottom": 2,
                        "paddingTop": 22,
                      },
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      Object {
                        "marginRight": 36,
                        "paddingRight": 8,
                      },
                    ]
                  }
                  testID="sign-up-input-password"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "height": 24,
                      "justifyContent": "center",
                      "position": "absolute",
                      "width": 24,
                    },
                    Object {
                      "right": 12,
                      "top": 14,
                    },
                  ]
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  hitSlop={
                    Object {
                      "bottom": 6,
                      "left": 6,
                      "right": 6,
                      "top": 6,
                    }
                  }
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "margin": 6,
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 18,
                          "height": 36,
                          "width": 36,
                        },
                        undefined,
                        Object {
                          "margin": 0,
                        },
                      ],
                    ]
                  }
                >
                  <View>
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 24,
                          },
                          Array [
                            Object {
                              "lineHeight": 24,
                              "transform": Array [
                                Object {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                          ],
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      Û∞àâ
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#6200ee",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 2,
                "marginHorizontal": 18,
                "marginTop": 30,
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 0.75,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 1.5,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                ]
              }
              testID="sign-up-button-create"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Object {
                      "width": 714,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        false,
                        Object {
                          "color": "#ffffff",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "paddingTop": 2,
                        },
                      ],
                    ]
                  }
                >
                  Create an account
                </Text>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            nativeID="animatedComponent"
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0,
                "bottom": 50,
                "elevation": 0,
                "minWidth": 64,
                "position": "absolute",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                ]
              }
              testID="sign-up-button-signin"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Object {
                      "width": 714,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        false,
                        Object {
                          "color": "#6200ee",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "color": "#777b",
                          "marginTop": 10,
                          "paddingHorizontal": 18,
                        },
                      ],
                    ]
                  }
                >
                  Already have an account? Sign in
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</RCTScrollView>
`;
